src/components/date-picker/date-picker-dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import {getMonthNames, getMonthFromOffset} from './date-picker-helpers';
+import {observable, action, computed} from 'mobx';
+import {observer} from 'mobx-react';
+import {DatePickerGrid} from './date-picker-grid';
+import styles from './date-picker.st.css';
+
+export interface DatePickerDropdownProps {
+    date: Date;
+    onChange (date: Date): void;
+}
+
+const monthNames = getMonthNames();
+
+@observer
+export class DatePickerDropdown extends React.Component<DatePickerDropdownProps, {}>{
+    @observable date: Date = this.props.date;
+
+    @action setDateTo (date: Date) {
+        this.date = date;
+    }
+
+    @action setDayTo (day: string) {
+        const date = new Date(this.date.getFullYear(), this.date.getMonth(), parseInt(day));
+        this.props.onChange(date);
+    }
+
+    @computed
+    get monthName (): string {
+        return monthNames[this.date.getMonth()];
+    }
+
+    @computed
+    get year (): number {
+        return this.date.getFullYear();
+    }
+
+    goToNextMonth: React.EventHandler<React.SyntheticEvent<HTMLSpanElement>> = (event: React.SyntheticEvent<HTMLSpanElement>) => {
+        event.preventDefault();
+        const nextMonth: Date = getMonthFromOffset(new Date(this.date.getFullYear(), this.date.getMonth(), 1), 1);
+        this.setDateTo(nextMonth);
+    };
+
+    goToPrevMonth: React.EventHandler<React.SyntheticEvent<HTMLSpanElement>> = (event: React.SyntheticEvent<HTMLSpanElement>) => {
+        event.preventDefault();
+        event.stopPropagation();
```

Comment:
This doesn't appear in the previous method. They should be symmetrical. Also, why is preventDefault() not enough?